Source/FullScreenPokemon.ts

```diff
@@ -1006,16 +1006,31 @@ module FullScreenPokemon {
         }
 
         /**
-         * Reacts to the select key being pressed. Toggles the use of the registered item
+         * Reacts to the select key being pressed. Toggles the use of the registered item.
          * 
          * @param thing   The triggering Character.
          * @param event   The original user-caused Event.
          * @todo Extend the use for any registered item, not just the bicycle.
-         * @todo If the item isn't able to be used in this area, display so in the game.
          */
         keyDownSelect(thing: ICharacter, event?: Event): void {
             thing.FSP.ModAttacher.fireEvent("onKeyDownSelect");
-            thing.FSP.toggleCycling(<IPlayer>thing);
+
+            if (thing.FSP.MenuGrapher.getActiveMenu()) {
+                return;
+            }
+
+            if (!thing.FSP.toggleCycling(<IPlayer>thing)) {
+                thing.FSP.MenuGrapher.createMenu("GeneralText", {
```

Comment:
Should this menu logic should be in its own function? Other select-triggered items also have similar or same dialogs right?